### PR TITLE
fix: [CDS-50690]: Showing Git commit as output NG

### PR DESCRIPTION
**Pull request by TarunUba1**
## Describe your changes
Adding Commit Id to as an execution outcome for NG.
## Testing
Screenshot 2023-03-03 at 12 02 46 AM
Screenshot 2023-03-03 at 12 18 43 AM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)

### CODE REVIEW

File: 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java [hunk 1]
```diff
@@ -580,7 +580,10 @@ public enum FeatureName {
   ENABLE_K8_BUILDS("FF for enabling kubernetes builds as an option", HarnessTeam.CI),
   PL_USER_DELETION_V2("Modularising user deletion flows separately for CG and NG ", HarnessTeam.PL),
   DISABLE_INSTANCE_STATS_JOB_CG("Disable Instance Stats Job from CG", HarnessTeam.CDP),
-  CG_K8S_MANIFEST_COMMIT_VAR(
+  CDS_CG_K8S_MANIFEST_COMMIT_VAR(
```
Comment (abosii-harness) [inline 956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java:583]: we should not change existing FF

File: 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenStep.java [hunk 2]
```diff
@@ -195,16 +195,18 @@ public StepResponse finalizeExecutionWithSecurityContext(Ambiance ambiance,
...
+        K8sBlueGreenOutcome.builder()
+            .releaseName(cdStepHelper.getReleaseName(ambiance, infrastructure))
+            .releaseNumber(k8sBGDeployResponse.getReleaseNumber())
+            .primaryServiceName(k8sBGDeployResponse.getPrimaryServiceName())
+            .stageServiceName(k8sBGDeployResponse.getStageServiceName())
+            .stageColor(k8sBGDeployResponse.getStageColor())
+            .primaryColor(k8sBGDeployResponse.getPrimaryColor())
+            .prunedResourceIds(
+                k8sStepHelper.getPrunedResourcesIds(pruningEnabled, k8sBGDeployResponse.getPrunedResourceIds()))
+            .manifest(((K8sExecutionPassThroughData) passThroughData).getK8sGitFetchInfo())
```
Comment (abosii-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenStep.java:208]: you already have  K8sExecutionPassThroughData executionPassThroughData, no need for casting here

File: 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryOutcome.java [hunk 3]
```diff
@@ -30,5 +32,6 @@ public class K8sCanaryOutcome implements Outcome, ExecutionSweepingOutput {
   Integer targetInstances;
   Integer releaseNumber;
   String canaryWorkload;
+  K8sGitFetchInfo manifest;
```
Comment (abosii-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryOutcome.java:35]: Let's get approval on the expression

File: 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java [hunk 4]
```diff
@@ -101,7 +103,10 @@
 import io.harness.steps.TaskRequestsUtils;
 import io.harness.supplier.ThrowingSupplier;
 import io.harness.tasks.ResponseData;
+import io.harness.utils.NGFeatureFlagHelperService;
 
+import software.wings.api.k8s.K8sGitFetchInfo;
```
Comment (abosii-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java:108]: Let's not reuse CG beans

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/git/GitFetchTaskNG.java [hunk 5]
```diff
@@ -103,8 +104,10 @@ public GitFetchResponse run(TaskParameters parameters) {
                       gitFetchFilesConfig.getIdentifier()),
                 White, Bold));
 
+        String k8ValuesLocation = gitFetchFilesConfig.getIdentifier();
```
Comment (abosii-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/git/GitFetchTaskNG.java:107]: This is not k8sValuesLocation but identifier

**Review by abosii-harness — CHANGES_REQUESTED** [review]

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 29 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenStep.java` line 208 at r1](https://reviewable.io/reviews/harness/harness-core/44629#-NPYBJxSlzFtEc4H7t0U-r1-208:-NPb7WwV9lHVROwuwAz2:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2ae11369a5da76bc8a7b9171e3718eb5453eba56/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sBlueGreenStep.java#L208)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

you already have  K8sExecutionPassThroughData executionPassThroughData, no need for casting here
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java` line 108 at r1](https://reviewable.io/reviews/harness/harness-core/44629#-NPYBJxSlzFtEc4H7t0a-r1-108:-NPbAjmAAIZvhvnfxfVP:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2ae11369a5da76bc8a7b9171e3718eb5453eba56/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java#L108)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Let's not reuse CG beans
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/git/GitFetchTaskNG.java` line 107 at r1](https://reviewable.io/reviews/harness/harness-core/44629#-NPYBJxSlzFtEc4H7t0d-r1-107:-NPb9B991Fq8C9TaeTNw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2ae11369a5da76bc8a7b9171e3718eb5453eba56/930-delegate-tasks/src/main/java/io/harness/delegate/task/git/GitFetchTaskNG.java#L107)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

This is not k8sValuesLocation but identifier
</blockquote></details>

Done.
___
*[`956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java` line 583 at r1](https://reviewable.io/reviews/harness/harness-core/44629#-NPYBJxSlzFtEc4H7t0l-r1-583:-NPbApis5y-NmuQApl7k:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2ae11369a5da76bc8a7b9171e3718eb5453eba56/956-feature-flag-beans/src/main/java/io/harness/beans/FeatureName.java#L583)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

we should not change existing FF
</blockquote></details>

Done.

**Review by abhinavsingh1196 — CHANGES_REQUESTED** [review]
> Reviewed 13 of 27 files at r1, 8 of 12 files at r2, 5 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 29 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java` line 680 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPpiO0k90zLTfJ950xk:-NPpiO0k90zLTfJ950xl:b74mf3q) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java#L680)):*
> ```Java
>       Map<String, String> gitFetchFilesConfigMap = gitFetchResponse.getFetchedCommitIdsMap();
>       gitFetchFilesConfigMap.forEach(
>           (String keys, String values) -> { variables.put(keys, K8sGitInfo.builder().commitId(values).build()); });
> ```

redundant braces {}
also, can be renamed to "key" and "value"



_Code quote:_
```Java
K8sGitFetchInfo k8sGitFetchInfo = K8sGitFetchInfo.builder().build();
```

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 13 of 27 files at r1, 9 of 12 files at r2, 5 of 7 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 3 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPq4Gbw7yS2Kzgb6Xg9:-NPq4Gbw7yS2Kzgb6XgA:b1ve8f2) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java#L26)):*
> ```Java
> @Builder
> @JsonTypeName("K8sGitFetchInfo")
> @TargetModule(HarnessModule._957_CG_BEANS)
> ```

Why we need target modules here? Do we need to implement sweepingOutput? This is a CG entity if I remember right
___
*[`960-api-services/src/main/java/io/harness/git/Constants.java` line 37 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPq4ry4DAFrWzkfbMrO:-NPq4ry5Bt-pOgoygVUe:b1l6w4r) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/960-api-services/src/main/java/io/harness/git/Constants.java#L37)):*
> ```Java
>   String COMMIT_MESSAGE = "Harness IO Git Sync.\n";
>   String EXCEPTION_STRING = "Exception: ";
>   String DEFAULT_FETCH_IDENTIFIER = "--default";
> ```

nit: better __DEFAULT

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 4 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryOutcome.java` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/44629#-NPYBJxSlzFtEc4H7t0V-r1-35:-NPqINgIA6RPTrChYK_D:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2ae11369a5da76bc8a7b9171e3718eb5453eba56/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sCanaryOutcome.java#L35)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Let's get approval on the expression
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java` line 680 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPpiO0k90zLTfJ950xk:-NPqIMqwBuMO-npwz-dH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/K8sStepHelper.java#L680)):*
<details><summary><i>Previously, abhinavsingh1196 (Abhinav Singh) wrote…</i></summary><blockquote>

redundant braces {}
also, can be renamed to "key" and "value"
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPq4Gbw7yS2Kzgb6Xg9:-NPqIP6Z2YczTWmxb7nH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java#L26)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Why we need target modules here? Do we need to implement sweepingOutput? This is a CG entity if I remember right
</blockquote></details>

Done.

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 7 files at r3, 1 of 3 files at r5, 1 of 1 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPq4Gbw7yS2Kzgb6Xg9:-NPqeHuB1qWAiLygqLBs:b-ulxiym) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java#L26)):*
<details><summary><i>Previously, TarunUba1 (Tarun Uba) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

What about implements SweepingOutput?

**Review by TarunUba1 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java` line 26 at r4](https://reviewable.io/reviews/harness/harness-core/44629#-NPq4Gbw7yS2Kzgb6Xg9:-NPqjxqF2QWWPmMjNpR4:b-p7hwuk) ([raw file](https://github.com/harness/harness-core/blob/6eef7c6f8b6dd8342ea311d6bf9f57bda412eaa9/127-cd-nextgen-entities/src/main/java/io/harness/cdng/k8s/K8sGitFetchInfo.java#L26)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

What about implements SweepingOutput?
</blockquote></details>

FIXED, my bad

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by abosii-harness** [comment]
Please provide a better PR title, this PR doesn't update FF name anymore

**Review by abhinavsingh1196 — APPROVED** [review]
> Reviewed 1 of 7 files at r3, 1 of 3 files at r5, 2 of 2 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44629)* status: :shipit: complete! ✓

**Pull request merged**